com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/AdapterUtils.java

```diff
@@ -287,4 +295,52 @@ public static String decodeURIComponent(String uri) {
             return uri;
         }
     }
+
+    /**
+     * Generate the classpath parameters to a temporary classpath.jar.
+     * @param classPaths - the classpath parameters
+     * @return the file path of the generate classpath.jar
+     * @throws IOException Some errors occur during generating the classpath.jar
+     */
+    public static Path generateClasspathJar(String[] classPaths) throws IOException {
+        List<String> classpathUrls = new ArrayList<>();
+        for (String classpath : classPaths) {
+            classpathUrls.add(AdapterUtils.toUrl(classpath));
+        }
+
+        Manifest manifest = new Manifest();
+        Attributes attributes = manifest.getMainAttributes();
+        attributes.put(Attributes.Name.MANIFEST_VERSION, "1.0");
+        // In jar manifest, the absolute path C:\a.jar should be converted to the url style file:///C:/a.jar
+        attributes.put(Attributes.Name.CLASS_PATH, String.join(" ", classpathUrls));
+        Path tempfile = Files.createTempFile("classpath_", ".jar");
+        JarOutputStream jar = new JarOutputStream(new FileOutputStream(tempfile.toFile()), manifest);
+        jar.close();
+
+        return tempfile;
+    }
+
+    /**
+     * Generate the classpath parameters to a temporary argfile file.
+     * @param classPaths - the classpath parameters
+     * @param modulePaths - the modulepath parameters
+     * @return the file path of the generated argfile
+     * @throws IOException Some errors occur during generating the argfile
+     */
+    public static Path generateArgfile(String[] classPaths, String[] modulePaths) throws IOException {
+        String argfile = "";
+        if (ArrayUtils.isNotEmpty(classPaths)) {
```

Comment:
Is the path separate cross-platform safety?

Reply:
do you mean `File.pathSeparator`? it's platform dependent.

Reply:
Yes, lots of hard code with path manipulation. Should aware of platform for path operations.